jsonnet-mode.el

```diff
@@ -651,20 +651,21 @@ TYPE is an opening paren-like character."
     (when-let ((output-window (get-buffer-window output-buffer-name t)))
       (quit-window nil output-window)
       (redisplay))
-    (with-current-buffer (get-buffer-create output-buffer-name)
-      (setq buffer-read-only nil)
-      (erase-buffer)
-      (let ((args (nconc jsonnet-command-options
-                         (cl-loop for dir in search-dirs
-                                  collect "-J"
-                                  collect dir)
-                         (list file-to-eval))))
-        (if (zerop (apply #'call-process jsonnet-command nil t nil args))
-            (progn
-              (when (fboundp 'json-mode)
-                (json-mode))
-              (view-mode))
-          (compilation-mode nil)))
+    (let ((cmd jsonnet-command)
+          (args (append jsonnet-command-options
+                        (cl-loop for dir in search-dirs
+                                collect "-J"
+                                collect dir)
+                       (list file-to-eval))))
+      (with-current-buffer (get-buffer-create output-buffer-name)
+       (setq buffer-read-only nil)
+       (erase-buffer)
+       (if (zerop (apply #'call-process cmd nil t nil args))
+           (progn
+             (when (fboundp 'json-mode)
+               (json-mode))
+             (view-mode))
+         (compilation-mode nil)))
       (goto-char (point-min))
       (display-buffer (current-buffer)
                       '((display-buffer-pop-up-window
```

Comment:
It was probably inadvertent, but `display-buffer` has been moved outside the body of the preceding `with-current-buffer` body; for better or worse, `display-buffer`'s argument in this instance is `(current-buffer)`, so we don't see the JSON output displayed in the other window. You can either move it back to the body of `with-current-buffer` or change its argument to `output-buffer-name`.

In any case, thanks for the PR!

Reply:
whoops sorry about that, I think we should keep the current flow for the time being.  Yeah it was an oversight.  I was doing some experiments with the `display-buffer` behaviour,  but I don't have anything of substance to contribute at this time.

I have pushed an updated commit with the correct scoping

Reply:
No worries!

WRT `display-buffer` behavior: I think the current approach is fine and (if I understand correctly) it can be overridden via `display-buffer-alist` or a related variable.

Reply:
it's really weird the behaviour i see,  when pressing `C-c C-c`

on one machine it hides the buffer, and is buried, it never returns (emacs 27.1)

on another machine the window closes then opens again (emacs 28), so it looks like windows are being re-arranged, it's not behaviour that i see from other compile modes.

So i'm going to do a bit of a dive into this and see if i can figure out what is going on,  i saw that the upstream compile mode uses `(display-buffer outbuf '(nil (allow-no-window . t)))` as their display function.

do you customize `display-buffer-alist` for this mode?

Reply:
Hmmm... interesting. I haven't customized `display-buffer-alist` in a way that should affect Jsonnet's compilation buffer. Regardless, I added new tests that _should_ [catch any unusual behavior](https://github.com/tminor/jsonnet-mode/blob/main/jsonnet-test.el#L73-L76) though I'm sure it could use some improvement!

We have CI configured to run on Emacs 25.1, 25.3 (not sure why I configured two different versions...), 26.3, 27.1, and snapshot. I'm not very familiar with Nix (or Emacs's release cycle), but if I understand correctly, the currently configured workflow bases "snapshot" on emacs-mirror/emacs@1283e1db which is Emacs 28.0.50. Are you using nativecomp with Emacs 28? [The config flags](https://github.com/purcell/nix-emacs-ci/blob/master/emacs.nix#L51-L61) unsurprisingly don't specify `--with-native-compilation` so that may explain the odd behavior.

Reply:
Perhaps this discussion warrants a new issue? If you can reliably reproduce the behavior by running `emacs -Q`, please open one and include the recipe. Thanks!

Reply:
Thanks for the reply, I'll reproduce it tonight and send in a report and possibly a fix, since usually i'll have to tinker to understand what is going on.